ts/modules/marker-clusters.ts

```diff
@@ -0,0 +1,1948 @@
+/* *
+ *
+ *  Marker clusters module.
+ *
+ *  (c) 2010-2019 Torstein Honsi
+ *
+ *  Author: Wojciech Chmiel
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import H from './../parts/Globals.js';
+import U from './../parts/Utilities.js';
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        type ClusterAlgorithms = (
+            'grid'|'kmeans'|'optimizedKmeans'
+        );
+        interface LayoutAlgorithmOptions {
+            type: (ClusterAlgorithms | Function);
+            gridSize: (number | string);
+            processedGridSize: number;
+            distance: (number | string);
+            processedDistance: number;
+            iterations?: number;
+            kmeansThreshold?: number;
+        }
+        interface StyleOptions {
```

Comment:
`StyleOptions`, and by extension the `cluster.style` option set, is not a good name. By convention, the options called `style` in Highcharts are CSS properties directly relayed to some element, usually (perhaps always) text elements or labels. The properties here are more general cluster presentation settings.

Reply:
I agree. If it does contain common styling properties so that `cluster.style` can be directly of type `Highcharts.CSSObject` without the interface above, then use a different name like `ThemeOptions`.

Reply:
Beside that you also should avoid common interface names, as all source files share the same `Highcharts` namespace in TypeScript. So in your case all new interface should start with `MarkerCluster…` prefix.